### PR TITLE
Remove ability to configure page size.

### PR DESCRIPTION
**Pull request by ragadeeshu**
Also adds a way to specify it via the PageCahceRule, so that it can be used for
testing.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/pagecache/ConfiguringPageCacheFactory.java [hunk 1]
```diff
@@ -43,6 +44,9 @@
     private final PageCacheTracer tracer;
     private final Log log;
     private PageCache pageCache;
+    private int configuredPageSize;
+
+    private static final int pageSize = getInteger( ConfiguringPageCacheFactory.class, "pageSize", 8192 );
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/pagecache/ConfiguringPageCacheFactory.java]: `static finals` should be in the top of the class.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/pagecache/ConfiguringPageCacheFactory.java [hunk 2]
```diff
@@ -51,6 +55,16 @@ public ConfiguringPageCacheFactory(
...
+    }
+
+    /**
+     * Sets the page size, used by the {@link StandalonePageCacheFactory} for testing purposes.
+     * @param pageSize The new page size.
+     */
+    public void setPageSize( int pageSize )
+    {
+        configuredPageSize = pageSize;
     }
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/pagecache/ConfiguringPageCacheFactory.java]: I'd do this juggling in a `TestConfiguringPageCacheFactory` that extends and overrides the original `ConfiguringPageCacheFactory`. The `PageCacheRule` can then use that when configuring its page cache.

**Review by chrisvest — CHANGES_REQUESTED** [review]
I'd like greater separation of product and testing features.

**Comment by ragadeeshu** [comment]
@chrisvest Like this?

**Review by chrisvest — APPROVED** [review]

**Comment by chrisvest** [comment]
Yep.

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java [hunk 3]
```diff
@@ -202,8 +201,7 @@ protected static Config getNeo4jConfig( Configuration config, Config dbConfig )
     {
         return new Config( stringMap( dbConfig.getParams(),
                 dense_node_threshold.name(), valueOf( config.denseNodeThreshold() ),
-                pagecache_memory.name(), valueOf( config.pageCacheMemory() ),
-                mapped_memory_page_size.name(), valueOf( config.pageSize() ) ),
+                pagecache_memory.name(), valueOf( config.pageCacheMemory() ) ),
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java:204]: Just noticed the `config` object still has its `pageSize()` method. Can that be removed?

Comment (ragadeeshu) [reply community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java:204]: Nice catch, will look into it.

Comment (ragadeeshu) [reply community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java:204]: @chrisvest removed

**Pull request merged**